### PR TITLE
Fixed memory leak

### PR DESCRIPTION
**Pull request by Peteck**
First of all, thanks for this amazing lib. The API looks amazing!

So I just started testing soundio, but I kept getting a visual leak detection. The leak was caused by `soundio_connect(soundio);`

### CODE REVIEW

**Comment by Peteck** [comment]
Maybe I need to put this free's somewhere else. I guess there can be cases this fix doesn't account for, but where I put them now fixed the problem I'm having.

File: src/wasapi.c [hunk 1]
```diff
@@ -486,8 +486,12 @@ static void deinit_refresh_devices(struct RefreshDevices *rd) {
     soundio_device_unref(rd->device_raw);
     if (rd->mm_device)
         IMMDevice_Release(rd->mm_device);
-    if (rd->default_render_device)
-        IMMDevice_Release(rd->default_render_device);
+	if (rd->default_render_device)
+	{
+		IMMDevice_Release(rd->default_render_device);
+		free(rd->default_capture_id);
```
Comment (wegylexy) [inline src/wasapi.c:492]:
```suggestion
```

File: src/wasapi.c [hunk 2]
```diff
@@ -486,8 +486,12 @@ static void deinit_refresh_devices(struct RefreshDevices *rd) {
...
+		free(rd->default_render_id);
+	}
     if (rd->default_capture_device)
         IMMDevice_Release(rd->default_capture_device);
```
Comment (wegylexy) [inline src/wasapi.c:496]:
```suggestion
    {
        IMMDevice_Release(rd->default_capture_device);
        free(rd->default_capture_id);
    }
```

**Comment by Peteck** [comment]
thank you, @wegylexy! :-)

**Comment by Peteck** [comment]
@andrewrk Could we soon merge this fix?

**Comment by wegylexy** [comment]
@andrewrk Do you also have time to evaluate the other pull requests? Especially those WASAPI ones as the current thing is partially broken on Windows 10 v2004.

**Comment by andrewrk** [comment]
No I won't be making another release for about 2 years, but then it should pick back up. The biggest problem with pull requests is testing. QA for this software is near impossible, we need actual hardware and half the time I ask people if they tested their code, they say "no, I just guessed this would work".

**Pull request merged**